kafka/src/main/java/org/apache/apex/malhar/kafka/KafkaSinglePortExactlyOnceOutputOperator.java

```
@@ -0,0 +1,369 @@
+  private final String KEY_SEPARATOR = "#";
+  private final String KEY_SERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
+  private final String VALUE_SERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
+
+  public final transient DefaultInputPort<T> inputPort = new DefaultInputPort<T>()
+  {
+    @Override
+    public void process(T tuple)
+    {
+      sendTuple(tuple);
+    }
+  };
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    super.setup(context);
+
+    this.operatorId = context.getId();
+    this.windowDataManager.setup(context);
+    this.appName = context.getValue(Context.DAGContext.APPLICATION_NAME);
+    this.key = appName + KEY_SEPARATOR + (new Integer(operatorId));
+    this.consumer = KafkaConsumerInit();
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    this.windowId = windowId;
+
+    if (windowId == windowDataManager.getLargestRecoveryWindow()) {
+      rebuildPartialWindow();
+    }
+  }
+
+  @Override
+  public void checkpointed(long windowId)
+  {
+  }
+
+  @Override
+  public void committed(long windowId)
+  {
+    try {
+      windowDataManager.deleteUpTo(operatorId, windowId);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void beforeCheckpoint(long windowId)
+  {
+  }
+
+  @Override
+  public void teardown()
+  {
+    consumer.close();
+    super.teardown();
+  }
+
+  @Override
+  public void endWindow()
+  {
+    if (!partialWindowTuples.isEmpty() && windowId > windowDataManager.getLargestRecoveryWindow()) {
+      throw new RuntimeException("Violates Exactly once. Not all the tuples received after operator reset.");
+    }
+
+    try {
+      this.windowDataManager.save(getPartitionsAndOffsets(), operatorId, windowId);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Don't swallow exception here